earthpy/spatial.py

```diff
@@ -463,7 +473,7 @@ def plot_rgb(arr, rgb=(0, 1, 2),
                            order with bands first""")
 
     # Index bands for plotting and clean up data for matplotlib
-    rgb_bands = arr[rgb]
+    rgb_bands = arr[rgb, :, :]
```

Comment:
This change I think fixes a bug in `plot_rgb`. Previously this line would return one value for the default value of the `rgb` argument, because a 3 dimension ndarray was being indexed using a length 3 tuple, e.g.,:

```python
>>> import numpy as np
>>> arr = np.arange(9).reshape((1, 3, 3))
>>> rgb = (0, 1, 2)  # the default value for the rgb parameter
>>> rgb_bands = arr[rgb] 
>>> rgb_bands
5
```

@lwasser please take a close look at this change and make sure I'm understanding correctly. I'm not sure whether this function is in active usage in your class, so maybe I've missed something and this is not actually a bug.

Reply:
hmmm definitely confused here. this function did work just fine before but i think i need to dig into it to understand what the bug is you are mentioned. i do remember seeing a deprecation of subsetting the way we were doing it...

Reply:
Fixed this IRL!